### ISSUE TITLE
MSBuild for .NET Core cannot create VSIX

### ISSUE THREAD

**Issue by couven92** [microsoft/msbuild#2653]
## TL;DR
When importing the targets file from the `Microsoft.VSSDK.BuildTools` NuGet package, MSBuild sets the `PrepareForRunDependsOn` property. This property is then later reset by the `Microsoft.Common.CurrentVersion.targets` file that is included in the SDK imports. This behaviour prevents MSBuild from creating a VSIX package after a successful build.

## Problem
In order to build a VSIX project using the .NET Core CLI I did the following:
* created a new .NET Core class library
* added a tooling package reference to the `Microsoft.VSSDK.BuildTools` NuGet package like this:  `<ItemGroup><DotNetCliToolReference Include="Microsoft.VSSDK.BuildTools" Version="15.1.192" /></ItemGroup>`
* Imported the VSSDK properties and targets using the `NugetPackageRoot` property like this:  
`<Import Project="$(NugetPackageRoot)Microsoft.VSSDK.BuildTools\15.1.192\build\Microsoft.VSSDK.BuildTools.props" />`  
`<Import Project="$(NugetPackageRoot)Microsoft.VSSDK.BuildTools\15.1.192\build\Microsoft.VSSDK.BuildTools.targets" />`  
`<Import Project="$(VSToolsPath)\VSSDK\Microsoft.VsSDK.targets" Condition="'$(VSToolsPath)' != ''" />`

When I run `dotnet build` the project build without error, but only produces a regular class library, it does not create a VSIX package.

I proceeded to run `dotnet msbuild /bl:MSBuild.Core.binlog` to debug the build process.

During the MSBuild evaluation phase I noticed the following
* In regular full Framework MSBuild, the targets to execute to create a VSIX after build are expanded into the `PrepareForRunDependsOn` property.
* In my setup for .NET Core, MSBuild also expands the `PrepareForRunDependsOn` property with the correct values
* MSBuild for .NET Core then overrides the `PrepareForRunDependsOn` from the `Microsoft.Common.CurrentVersion.targets` file included in the SDK and sets it back to `CopyFilesToOutputDirectory`

This indicates an error in the MSBuild targets evaluation order. Am I doing something fundamentally wrong? How can I modify the targets evaluation order correctly?

### Full examplary binary log files
All MSBuild relevant files should be embedded in the following binary log files. For comparison I have provided logs from both a full Visual Studio with the VSSDK workload installed and the .NET Core builds.
* [MSBuild.binlog](https://github.com/couven92/auran/raw/msbuild-log/MSBuild.binlog)
* [MSBuild.Core.binlog](https://github.com/couven92/auran/raw/msbuild-log/MSBuild.Core.binlog)

### MSBuild Version
For full Framework:
```
> MSBuild /version
Microsoft (R) Build Engine version 15.4.8.50001 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

15.4.8.50001
```

For .NET Core:
```
> dotnet --info
.NET Command Line Tools (2.0.2)

Product Information:
 Version:            2.0.2
 Commit SHA-1 hash:  a04b4bf512

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.2\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d

> dotnet msbuild /version
dotnet msbuild /version
Microsoft (R) Build Engine version 15.4.8.50001 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

15.4.8.50001
```

**Comment by rainersigwald** [comment]
Team triage: closing due to age. If you're still seeing this problem, please let us know and we can reactivate it.
